authority/tls.go

```diff
@@ -180,34 +189,48 @@ func (a *Authority) Renew(oldCert *x509.Certificate) ([]*x509.Certificate, error
 		PolicyIdentifiers:           oldCert.PolicyIdentifiers,
 	}
 
-	// Copy all extensions except for Authority Key Identifier. This one might
-	// be different if we rotate the intermediate certificate and it will cause
-	// a TLS bad certificate error.
+	// Copy all extensions except:
+	//	1. Authority Key Identifier - This one might be different if we rotate the intermediate certificate
+	//					and it will cause a TLS bad certificate error.
+	//	2. Subject Key Identifier - This should be calculated for the public key passed to this function.
 	for _, ext := range oldCert.Extensions {
-		if !ext.Id.Equal(oidAuthorityKeyIdentifier) {
+		if (!ext.Id.Equal(oidAuthorityKeyIdentifier)) && (!ext.Id.Equal(oidSubjectKeyIdentifier)) {
 			newCert.ExtraExtensions = append(newCert.ExtraExtensions, ext)
 		}
+		if ext.Id.Equal(oidSubjectKeyIdentifier) {
+			pubBytes, err := x509.MarshalPKIXPublicKey(pk)
+			if err != nil {
+				return nil, errs.Wrap(http.StatusInternalServerError, err,
+					"authority.RenewOrRekey; error marshaling public key", opts...)
+			}
+			hash := sha1.Sum(pubBytes)
+			skiExtension := pkix.Extension{
+				Id:    oidSubjectKeyIdentifier,
+				Value: append([]byte{4, 20}, hash[:]...),
+			}
+			newCert.ExtraExtensions = append(newCert.ExtraExtensions, skiExtension)
+		}
 	}
 
 	leaf, err := x509util.NewLeafProfileWithTemplate(newCert, a.x509Issuer, a.x509Signer)
 	if err != nil {
-		return nil, errs.Wrap(http.StatusInternalServerError, err, "authority.Renew", opts...)
+		return nil, errs.Wrap(http.StatusInternalServerError, err, "authority.RenewOrRekey", opts...)
```

Comment:
This can be perfectly just Rekey, as the logs would tell us if the error was coming from the renew or rekey method, another option would be to encapsulate all the logic in a PRIVATE method and pass the method identifier for logging.

Reply:
I don't understand the second option. Meanwhile i will go with just renaming to Rekey everywhere.